### PR TITLE
Our first PR!

### PR DESCRIPTION
**Pull request by Eleven1Liu**
This is the first pull request.

### CODE REVIEW

**Review by henryyang42 — DISMISSED** [review]

File: data_utils.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            labels = f.readlines()
+            dataset[split]['label'] = [label.split() for label in labels]
+        dataset[split]['index'] = list(range(len(dataset[split]['text'])))
+    return dataset
+
+
+@log.enter('load_dataset')
+def load_dataset(config):
+    """Preparing TextDataset from raw data."""
+    data_dir = os.path.join(config['data_dir'], config['data_name'])
```
Comment (henryyang42) [inline data_utils.py:93]: Unify all `config['xxx'] syntax to `config.xxx`?

File: main.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    parser.add_argument('--monitor_metrics', nargs='+', default=['P@1', 'P@3', 'P@5'], help='Metrics to monitor while validating (default: %(default)s)')
+    parser.add_argument('--val_metric', default='P@1', help='The metric to monitor for early stopping (default: %(default)s)')
+
+    # pretrained vocab / embeddings
+    parser.add_argument('--vocab_file', type=str, help='Path to a file holding vocabuaries (default: %(default)s)')
+    parser.add_argument('--embed_file', type=str, help='Path to a file holding pre-trained embeddings (default: %(default)s)')
+
+    # others
+    parser.add_argument('--cpu', action='store_true', help='Disable CUDA')
+    parser.add_argument('--display_iter', type=int, default=100, help='Log state after every n epochs (default: %(default)s)')
```
Comment (henryyang42) [inline main.py:70]: epochs -> steps

File: main.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    parser.add_argument('--data_workers', type=int, default=1, help='Use multi-cpu core for data pre-processing (default: %(default)s)')
+    parser.add_argument('--eval', action='store_true', help='Only run evaluation on the test set (default: %(default)s)')
+    parser.add_argument('--load_checkpoint', help='The checkpoint to warm-up with (default: %(default)s)')
```
Comment (henryyang42) [inline main.py:73]: ``load_checkpoint`` is more like an action to me. Rename to, for example `checkpoint_path`, to match the description `The checkpoint to warm-up with`?

File: data_utils.py (hunk 1 above)
Comment (skgg) [reply data_utils.py:93]: Some hyperparameter search tools may require configs as a dictionary.
Correct me if I'm wrong.

**Review by skgg — APPROVED** [review]

**Review by skgg — APPROVED** [review]

**Review by henryyang42 — APPROVED** [review]
